source/dem/dem.cc

```diff
@@ -627,30 +627,34 @@ DEMSolver<dim>::write_output_results()
                             group_files,
                             mpi_communicator);
 
-  // Write background grid
-  DataOut<dim> background_data_out;
-
-  background_data_out.attach_dof_handler(background_dh);
-
-  // Attach the solution data to data_out object
-  Vector<float> subdomain(triangulation.n_active_cells());
-  for (unsigned int i = 0; i < subdomain.size(); ++i)
-    subdomain(i) = triangulation.locally_owned_subdomain();
-  background_data_out.add_data_vector(subdomain, "subdomain");
-
-  const std::string grid_solution_name =
-    parameters.simulation_control.output_name + "-grid";
-
-  background_data_out.build_patches();
-
-  write_vtu_and_pvd<dim>(grid_pvdhandler,
-                         background_data_out,
-                         folder,
-                         grid_solution_name,
-                         time,
-                         iter,
-                         group_files,
-                         mpi_communicator);
+  if (parameters.post_processing.write_grid == true)
```

Comment:
Do you think there is a way that this grid could be merged with the lagrangian post-processing or you think this is too difficult? E.g. if we have lagrangian post-processing the info would just be added on top of this grid